program-error/src/lib.rs

```diff
@@ -125,17 +153,18 @@ impl fmt::Display for ProgramError {
 pub trait PrintProgramError {
```

Comment:
Wonder if we should deprecate this trait and/or put it behind the `solana-msg` feature?

cc: @joncinque

Reply:
This is a breaking change anyway, so we can just remove it, but let's do it in a different PR, this one is ready to go on my side.